From 197fc35b3d9d6f696dfb88cf4aed39035563066b Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 27 Jun 2017 15:07:14 +0100
Subject: [PATCH 149/215] SQUASH: mmc: Apply ERASE_BROKEN quirks correctly

Squash with: mmc: Add MMC_QUIRK_ERASE_BROKEN for some cards

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/mmc/core/quirks.h | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index 05c8d73..c9d5d64 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -94,12 +94,9 @@
 	 *  On some Kingston SD cards, multiple erases of less than 64
 	 *  sectors can cause corruption.
 	 */
-	MMC_FIXUP("SD16G", 0x41, 0x3432, add_quirk_mmc,
-		  MMC_QUIRK_ERASE_BROKEN),
-	MMC_FIXUP("SD32G", 0x41, 0x3432, add_quirk_mmc,
-		  MMC_QUIRK_ERASE_BROKEN),
-	MMC_FIXUP("SD64G", 0x41, 0x3432, add_quirk_mmc,
-		  MMC_QUIRK_ERASE_BROKEN),
+	MMC_FIXUP("SD16G", 0x41, 0x3432, add_quirk, MMC_QUIRK_ERASE_BROKEN),
+	MMC_FIXUP("SD32G", 0x41, 0x3432, add_quirk, MMC_QUIRK_ERASE_BROKEN),
+	MMC_FIXUP("SD64G", 0x41, 0x3432, add_quirk, MMC_QUIRK_ERASE_BROKEN),
 
 	END_FIXUP
 };
-- 
1.9.1

